openmp/runtime/src/CMakeLists.txt

```diff
@@ -174,17 +174,28 @@ if(NOT WIN32)
 endif()
 
 # Add the OpenMP library
-libomp_get_ldflags(LIBOMP_CONFIGURED_LDFLAGS)
 
+# First, create an OBJECT library with all the runtime sources.
+# This allows both the main library and unit tests to link against the same
+# compiled objects.
+add_library(omp_objects OBJECT ${LIBOMP_SOURCE_FILES})
```

Comment:
While I don't think it is elegant to introduce an object library, I haven't come up with a better solution to have unittests access to non-exported symbols.

The comment above could be improved: Shared/static libraries also "link" to the same compiled objects. It is just that the unittests also need access to private symbols not visible to outside the shared library.